platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -1008,7 +1024,16 @@ - (id)mgl_jsonExpressionObject {
                     [attributes addObject:jsonObject];
                 }
                 if (attributedExpression.attributes) {
-                    [attributes addObject:attributedExpression.attributes];
+                    NSMutableDictionary *attributedDictionary = [NSMutableDictionary dictionaryWithDictionary:attributedExpression.attributes];
+                    if (attributedDictionary[MGLFontNamesAttribute]) {
+                        attributedDictionary[MGLFontNamesAttribute] = @[@"literal", attributedDictionary[MGLFontNamesAttribute]];
+                    }
+                    if (attributedDictionary[MGLFontColorAttribute] && [attributedDictionary[MGLFontColorAttribute] isKindOfClass:[MGLColor class]]) {
+                        MGLColor *color = attributedDictionary[MGLFontColorAttribute];
+                        NSExpression *colorExpression = [NSExpression expressionForConstantValue:color];
+                        attributedDictionary[MGLFontColorAttribute] = colorExpression.mgl_jsonExpressionObject;
```

Comment:
This is a bit of a roundabout way to create a JSON expression object for a color literal. Factor this code out into an `MGLColor.mgl_jsonExpressionObject` property:

https://github.com/mapbox/mapbox-gl-native/blob/c2fe532b9bbc9db9b2bbab109f0f47f7cc9584d7/platform/darwin/src/NSExpression+MGLAdditions.mm#L982-L986